protocol/block.go

```diff
@@ -88,6 +88,17 @@ func (c *Chain) GenerateBlock(ctx context.Context, prev *legacy.Block, snapshot
 			continue
 		}
 
+		// Filter out transactions that are not yet valid, or no longer
+		// valid, per the block's timestamp.
+		if tx.Tx.MinTimeMs > b.TimestampMS {
+			// TODO(bobg): log this?
+			continue
+		}
+		if tx.Tx.MaxTimeMs < b.TimestampMS {
+			// TODO(bobg): log this?
```

Comment:
My vote would be nah, don't log it, since we surface this to the client more directly. (Either by the "expired" message in this PR or by a timeout on /submit if it's too far in the future.)